frontend/components/commons/forms/questions-form/fields/shared-components/label-selection/LabelSelection.component.vue

```diff
@@ -128,6 +140,17 @@ export default {
   },
   methods: {
     onSelect({ id, is_selected }) {
+      this.multiple ||
+        this.tranformOptionsForMonoSelectionAndEmitNewOptions({
+          id,
+          is_selected,
+        });
+    },
+    tranformOptionsForMonoSelectionAndEmitNewOptions({ id, is_selected }) {
+      this.factoryOptionsForMonoSelection({ id, is_selected });
+      this.$emit("on-change", this.options);
+    },
+    factoryOptionsForMonoSelection({ id, is_selected }) {
```

Comment:
Do we need all this refactor?  Maybe the change is easier to understand if we just add a condition related to the `multiple` prop:

```suggestion
      if (!!this.multiple) {
        this.options.map((option) => {
           option.is_selected = option.id === id ? is_selected : false;
           return option;
        });
        this.$emit("on-change", this.options);        
      }
```